subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -999,13 +999,20 @@ static int handle_sensor_status(struct bt_mesh_model *model, struct bt_mesh_msg_
 
 		/* Occupancy sensor */
 
-		/* OCC_MODE must be enabled for the occupancy sensors to be
-		 * able to turn on the light:
+		/* Occupancy event shall not have any affect in "Fade On" state
+		 * or "Fade Standby Manual" state. OCC_MODE must be enabled for
+		 * the occupancy sensors to be able to turn on the light when in
+		 * "Standby" state.
 		 */
-		if (srv->state == LIGHT_CTRL_STATE_STANDBY &&
-		    !atomic_test_bit(&srv->flags, FLAG_OCC_MODE) &&
-		    !(atomic_test_bit(&srv->flags, FLAG_TRANSITION) &&
-		      !atomic_test_bit(&srv->flags, FLAG_MANUAL))) {
+		if ((srv->state == LIGHT_CTRL_STATE_STANDBY &&
+			((!atomic_test_bit(&srv->flags, FLAG_OCC_MODE) &&
+			  !atomic_test_bit(&srv->flags, FLAG_TRANSITION) &&
+			  !atomic_test_bit(&srv->flags, FLAG_MANUAL))
+			  ||
+			  (atomic_test_bit(&srv->flags, FLAG_TRANSITION) &&
+			   atomic_test_bit(&srv->flags, FLAG_MANUAL)))) ||
+		    (srv->state == LIGHT_CTRL_STATE_ON &&
+		     atomic_test_bit(&srv->flags, FLAG_TRANSITION))) {
```

Comment:
The Occupancy On event can not be triggered in Fade On and Fade Standby Manual states, which correspond to the following flags:
| Spec State | Implementation state | Flags |
| --- | --- | --- |
| Fade On | LIGHT_CTRL_STATE_ON | FLAG_TRANSITION |
| Fade Manual Auto | LIGHT_CTRL_STATE_STANDBY | FLAG_TRANSITION && FLAG_MANUAL |

`FLAG_OCC_MODE` flag corresponds to Auto Occupancy.
So perhaps it should be like this?
```suggestion
		if (!atomic_test_bit(&srv->flags, FLAG_OCC_MODE) ||
		    (srv->state == LIGHT_CTRL_STATE_STANDBY &&
		     atomic_test_bit(&srv->flags, FLAG_TRANSITION) && 
		     atomic_test_bit(&srv->flags, FLAG_MANUAL)) ||
		    (srv->state == LIGHT_CTRL_STATE_ON &&
		     atomic_test_bit(&srv->flags, FLAG_TRANSITION))) {
```

Reply:
> Umm, may be. But, Fade Manual Auto? There is no such state.

Sorry, type, should have been `Fade Standby Manual` of course.

Reply:
From your suggestion:
`/* If in Run state stay in the same state (basically, do not process Occupancy). */`
seems not correct as Occupancy On event is present in the 6.5.2.8.

Reply:
The Occupancy conditions is only valid when in the spec standby state (in our code that means (standby, transition false, manual false), so the if the occ_mode is false this will remove occupancy for spec state Fade Standby Auto. The second state will remove occupancy from spec run state, as code on state both represent Fade On and Run state. Fade On does not have occupancy event while run state has occupancy. The third state represent Fade Standby Auto, but if OCC_Mode is false this condition will be overrun. This is why I made the if case specific to represent the state machine in the spec. I will add better comment to each state to explain it better.

Reply:
Unless I'm missing something, Occupancy On event is still present in Standy, Fade, Prolong and Fade Standby Auto states. It is the Condition that was updated. But, only Auto Occupancy had a special case in Standby state, in other states there was a dash line in the Condition column meaning that all Occupancy On conditions could be matched. So now, since other Occupancy Conditions were removed, for the states there is no change really except that Occupancy On event can trigger transition only if Auto Occupancy is enabled.

Reply:
> seems not correct as Occupancy On event is present in the 6.5.2.8.
Sorry, right. That comment and condition gets removed.

Reply:
There is 3 states Occupancy On event should not have any effect, when in Standby and Auto Occupancy is false, Fade On, Fade Standby Manual state. That is what I have added specifically in the cases. (And of course Off state, but that is in the top of the function.)

Reply:
I think this is a bit complicated. 

This IMO is difficult to understand:
```
((!atomic_test_bit(&srv->flags, FLAG_OCC_MODE) &&
  !atomic_test_bit(&srv->flags, FLAG_TRANSITION) &&
  !atomic_test_bit(&srv->flags, FLAG_MANUAL))
 ||
/* According to the Mesh Model Specification section
 * 6.2.5.12 specifications FADE STANDBY MANUAL state,
 * the Occupancy On event should not be processed.
 */
(atomic_test_bit(&srv->flags, FLAG_TRANSITION) &&
 atomic_test_bit(&srv->flags, FLAG_MANUAL)))) ||
```

This I think is more readable:
```
/* Auto Occupancy is disabled */
!atomic_test_bit(&srv->flags, FLAG_OCC_MODE) ||
/* Fade Standby Manual state */
(srv->state == LIGHT_CTRL_STATE_STANDBY &&
 atomic_test_bit(&srv->flags, FLAG_TRANSITION) && 
 atomic_test_bit(&srv->flags, FLAG_MANUAL)) ||
/* Fade On state */
(srv->state == LIGHT_CTRL_STATE_ON &&
 atomic_test_bit(&srv->flags, FLAG_TRANSITION))) {
```